src/sagas/nanoContract.js

```diff
@@ -0,0 +1,106 @@
+import {
+  ncApi
+} from '@hathor/wallet-lib';
+import {
+  takeEvery,
+  select,
+  all,
+  put,
+  call,
+} from 'redux-saga/effects';
+import { t } from 'ttag';
+import { NanoRequest404Error } from '@hathor/wallet-lib/lib/errors';
+import { STORE } from '../store';
+import {
+  nanoContractRegisterFailure,
+  nanoContractRegisterSuccess,
+  onExceptionCaptured,
+  types,
+} from '../actions';
+
+export const failureMessage = {
+  alreadyRegistered: t`Nano Contract already registered.`,
+  walletNotReadyError: t`Wallet is not ready yet to register a Nano Contract.`,
+  addressNotMine: t`The informed address does not belong to the wallet.`,
+  nanoContractStateNotFound: t`Nano Contract not found.`,
+  nanoContractStateFailure: t`Error while trying to get Nano Contract state.`,
+};
+
+/**
+ * Calls Nano Contract API to retrieve Nano Contract state.
+ * @param {string} ncId Nano Contract ID
+ * @returns {{
+ *   ncState?: Object,
+ *   error?: Error,
+ * }} Returns either an object containing ncState or an error.
+ */
+export async function getNanoContractState(ncId) {
+  try {
+    const state = await ncApi.getNanoContractState(ncId);
+    return { ncState: { ...state } };
+  } catch (err) {
+    return { error: err };
```

Comment:
Why should the error handling mechanism for this method be different from all other methods in the code?

I don't see any advantage in doing this, it just increases complexity and cognitive overhead imo

I would suggest to just let it throw and let whoever is calling it handle the error

Reply:
I believe try/catch breaks the code reading, especially when it is part of a script. That's why I often isolates it in a method call. But we can trade better reading for consistency.

Reply:
Done at https://github.com/HathorNetwork/hathor-wallet-mobile/pull/427/commits/df39218403f47bd38fc0897ea98eb52c6429feff